src/main/java/org/openhab/tools/analysis/checkstyle/BuildPropertiesCheck.java

```diff
@@ -0,0 +1,194 @@
+
+import org.eclipse.pde.core.build.IBuild;
+import org.eclipse.pde.core.build.IBuildEntry;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+
+/**
+ * Checks if a build.properties file is valid.
+ *
+ * @author Petar Valchev - Initial implementation
+ * @author Svilen Valkanov - Changes in the logic for the source property
+ *
+ */
+public class BuildPropertiesCheck extends AbstractStaticCheck {
+    private static final String PROPERTIES_EXTENSTION = "properties";
+    private static final String BIN_INCLUDES_PROPERTY_NAME = "bin.includes";
+    private static final String OUTPUT_PROPERTY_NAME = "output..";
+    private static final String SOURCE_PROPERTY_NAME = "source..";
+
+    private static final String MISSING_PROPERTY_MSG = "Missing %s property in the %s file.";
+    private static final String MISSING_VALUE_MSG = "Property  %s in the %s file is missing value: ";
+
+    private static final String BUILD_PROPERTIES_FILE_NAME = "build." + PROPERTIES_EXTENSTION;
+
+    private static final String MISSING_BIN_INCLUDES_PROPERTY_MSG = String.format(MISSING_PROPERTY_MSG,
+            BIN_INCLUDES_PROPERTY_NAME, BUILD_PROPERTIES_FILE_NAME);
+
+    private static final String MISSING_BIN_INCLUDES_VALUE_MSG = String.format(MISSING_VALUE_MSG,
+            BIN_INCLUDES_PROPERTY_NAME, BUILD_PROPERTIES_FILE_NAME);
+    private static final String MISSING_OUTPUT_VALUE_MSG = String.format(MISSING_VALUE_MSG, OUTPUT_PROPERTY_NAME,
+            BUILD_PROPERTIES_FILE_NAME);
+    private static final String MISSING_SRC_VALUE_MSG = String.format(MISSING_VALUE_MSG, SOURCE_PROPERTY_NAME,
+            BUILD_PROPERTIES_FILE_NAME);
+
+    private static final String EMPTY_FILE_MSG = String.format("Empty %s file", BUILD_PROPERTIES_FILE_NAME);
+
+    // Used for configuration properties
+    private List<String> expectedBinIncludesValues;
+    private List<String> expectedOutputValues;
+    private List<String> possibleSourceValues;
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public BuildPropertiesCheck() {
+        setFileExtensions(PROPERTIES_EXTENSTION);
+    }
+
+    // Configuration property for the expected values for
+    // the bin.includes property in the build.properties file.
+    public void setExpectedBinIncludesValues(String[] binIncludesValues) {
+        this.expectedBinIncludesValues = Arrays.asList(binIncludesValues);
+    }
+
+    // Configuration property for the expected values for
+    // the output property in the build.properties file.
+    public void setExpectedOutputValues(String[] outputValues) {
+        this.expectedOutputValues = Arrays.asList(outputValues);
+    }
+
+    // Configuration property for the possible values for
+    // the source property in the build.properties file.
+    // Some of the possible values will be filtered
```

Comment:
Found it, but I am not sure whether the comment is clear enough, it confused me more than it explained things to me.